scripts/cbmc_json_parser.py

```diff
@@ -255,6 +258,129 @@ def filter_properties(properties, message):
             filtered_properties.append(property)
     return filtered_properties, removed_properties
 
+class CProverCheck:
+    """ Represents a CProverCheck and provides methods to replace the check. """
+
+    def __init__(self, msg, new_msg=None):
+        self.original = msg
+        self.kani_msg = new_msg if new_msg else msg
+
+    def matches(self, msg):
+        return self.original in msg
```

Comment:
I need a substring matching. The purpose of this change is to remove the parts of the message that contains some expressions. I could use regex, but I thought it was a bit too much for now. I'll add an example of replacement we're making in the comments and hopefully it will clarify the intended behaviour.